### PR TITLE
RFC Proposal: Empty-scalars not allowed as keys

### PR DESCRIPTION
**Pull request by ingydotnet**
https://github.com/yaml/yaml-spec/blob/RFC-no-empty-key/rfc/RFC-no-empty-key.md

### CODE REVIEW

**Comment by pantoniou** [comment]
Looks good to me in general.

`- !null "": foo` means that we explicitly define that an empty scalar as equivalent to `null` or that *anything* with a !null tag is null too.

For instance is `- !null "null" : foo` the same as `- !null "": foo`

File: rfc/RFC-no-empty-key.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+You can also use these "empty" values as mapping keys.
+This means that `:` is same as `null: null`.
+
+Using a null value as a mapping key is a rare thing for a program to do (if it is even allowed by the programming language).
+
+Under this RFC you can still have a null key, but it needs to happen explicitly like in these examples:
+```
+- null: foo
+- !!null: foo
```
Comment (perlpunk) [inline rfc/RFC-no-empty-key.md:49]: This should actually be:
```
- !!null : foo
```
Otherwise the colon is part of the tag.
Same for `*null: foo` below

**Comment by perlpunk** [comment]
> `- !null "": foo` means that we explicitly define that an empty scalar as equivalent to `null` or that _anything_ with a !null tag is null too.

It's a local `!null` tag, not `!!null`.

File: rfc/RFC-no-empty-key.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+Using a null value as a mapping key is a rare thing for a program to do (if it is even allowed by the programming language).
+
+Under this RFC you can still have a null key, but it needs to happen explicitly like in these examples:
+```
+- null: foo
+- !!null : foo
+- !null "": foo
+- key: &null !null
+  *null: foo
```
Comment (perlpunk) [inline rfc/RFC-no-empty-key.md:52]:
```
  *null : foo
```
The alias needs a space before the colon

**Pull request merged**